File: js/without-di/createUser.js
const User = require('./User')
const UserRepository = require('./userRepository')
const MailService = require('./mailService')

const createUser = async (userData) => {
  const userRepository = UserRepository()
  const mailService = MailService()

  const newUser = User.build(userData)

  const user = await userRepository.save(newUser)

  mailService.sendConfirmEmail(user)

  return user
}

module.exports = createUser

File: ruby/src/create_user.rb
require_relative '../import.rb'
require_relative 'user.rb'

class CreateUser
  # usando dry-inject
  include Import['user_database']

  # caso for usar container manual ou DIY
  # def initialize(user_database:)
  #   @user_database = user_database
  # end

  def call(**user_data)
    user = User.new(**user_data)

    @user_database.save(user)
  end
end
